src/Random/InverseTransformSampling.h

```diff
@@ -0,0 +1,214 @@
+            fill_random(Distribution dist_, view_type x_, GeneratorPool rand_pool_, Vector<T, Dim> umin_, Vector<T, Dim> umax_)
+            : dist(dist_)
+            , x(x_)
+            , rand_pool(rand_pool_)
+            , umin_m(umin_)
+            , umax_m(umax_){}
+            KOKKOS_INLINE_FUNCTION void operator()(const size_t i) const {
+                typename GeneratorPool::generator_type rand_gen = rand_pool.get_state();
+                
+                value_type u = 0.0;
+                for (unsigned d = 0; d < Dim; ++d) {
+                    u       = rand_gen.drand(umin_m[d], umax_m[d]);
+                    
+                    // first guess for Newton-Raphson
+                    x(i)[d] = dist.estimate(u, d);
+                    
+                    // solve
+                    ippl::random::detail::NewtonRaphson<value_type> solver;
+
+                    solver.solve(dist, d, x(i)[d], u);
+
+                    rand_pool.free_state(rand_gen);
+                }
+            }
+        };
+        KOKKOS_INLINE_FUNCTION unsigned int getLocalNum() const { return nlocal_m; }
+        void generate(view_type view, Kokkos::Random_XorShift64_Pool<> rand_pool64) {
+            Kokkos::parallel_for(nlocal_m, fill_random<Kokkos::Random_XorShift64_Pool<>>(dist, view, rand_pool64, umin, umax));
+            Kokkos::fence();
+        }
+    private:
+        unsigned int nlocal_m;
+    };
+
+    template <typename T, unsigned Dim, unsigned DimP, typename PDF, typename CDF, typename ESTIMATE>
+    class Distribution {
+    public:
+       T par_m[DimP];
+
+       KOKKOS_INLINE_FUNCTION Distribution(const T *par_) {
+            for(unsigned int i=0; i<DimP; i++){
+                par_m[i] = par_[i];
+            }
+       }
+       PDF pdf_m;
+       CDF cdf_m;
+       ESTIMATE estimate_m;
+       KOKKOS_INLINE_FUNCTION T pdf(T x, unsigned int d) const{
+          return pdf_m(x, d, par_m);
+       }
+       KOKKOS_INLINE_FUNCTION T cdf(T x, unsigned int d) const{
+          return cdf_m(x, d, par_m);
+       }
+       KOKKOS_INLINE_FUNCTION T estimate(T x, unsigned int d) const{
+          return estimate_m(x, d, par_m);
+       }
+       KOKKOS_INLINE_FUNCTION T obj_func(T x, unsigned int d, T u) const{
+            return cdf(x, d) - u;
+       }
+       KOKKOS_INLINE_FUNCTION T der_obj_func(T x, unsigned int d) const{
+            return pdf(x, d);
+       }
+       KOKKOS_INLINE_FUNCTION T full_pdf(ippl::Vector<T, Dim> x) const{
+          T total_pdf = 1.0;
+          for(unsigned int d=0; d<Dim; d++){
+             total_pdf *= pdf(x[d], d);
+          }
+          return total_pdf;
+       }
+    };
+
+    template<typename T>
+    KOKKOS_FUNCTION T normal_cdf_func(T x, T mean, T stddev) {
+      return 0.5 * (1 + Kokkos::erf((x - mean) / (stddev * Kokkos::sqrt(2.0))));
+    }
+    template<typename T>
+    KOKKOS_FUNCTION T normal_pdf_func(T x, T mean, T stddev) {
+      T pi = Kokkos::numbers::pi_v<T>;
```

Comment:
consider to use `const` whenever it is possible. Is there a reason why you introduce `pi` as a variable?

Reply:
done. I'll try to use const when possible.